migrator/src/main/java/com/google/phonenumbers/migrator/MigrationJob.java

```diff
@@ -167,4 +206,155 @@ private DigitSequence getMigratedValue(String staleString, String oldFormat, Str
     }
     return DigitSequence.of(migratedValue.toString());
   }
+
+  /**
+   * Given a list of {@link MigrationResult}'s, returns a map detailing which migrations resulted in
+   * valid phone numbers based on the given rangesTable data. The map will contain to entries; an
+   * entry with the key 'Valid' with a list of the valid migrations and an entry with the key
+   * 'Invalid', with a list of the invalid migrations from the overall list.
+   */
+  private ImmutableMap<String, ImmutableList<MigrationResult>> verifyMigratedNumbers(
+      ImmutableList<MigrationResult> migrations) {
+    ImmutableList.Builder<MigrationResult> validMigrations = ImmutableList.builder();
+    ImmutableList.Builder<MigrationResult> invalidMigrations = ImmutableList.builder();
+
+    RangeTree validRanges = RangesTableSchema.toRangeTable(rangesTable).getAllRanges();
+    for (MigrationResult migration : migrations) {
+      DigitSequence migratedNum = migration.getMigratedNumber();
+      if (migratedNum.length() <= countryCode.length()) {
+        invalidMigrations.add(migration);
+        continue;
+      }
+      if(validRanges.contains(migratedNum.last(migratedNum.length() - countryCode.length()))) {
+        validMigrations.add(migration);
+      } else {
+        invalidMigrations.add(migration);
+      }
+    }
+    return ImmutableMap.of("Valid", validMigrations.build(), "Invalid", invalidMigrations.build());
+  }
+
+  /**
+   * Represents the results of a migration when calling {@link #getMigrationReportForCountry()}
+   * or {@link #getMigrationReportForRecipe(RangeKey)}
+   */
+  final class MigrationReport {
+    private final ImmutableList<MigrationEntry> untouchedEntries;
+    private final ImmutableList<MigrationResult> validMigrations;
+    private final ImmutableList<MigrationResult> invalidMigrations;
+
+    private MigrationReport(Stream<MigrationEntry> untouchedEntries,
+        ImmutableMap<String, ImmutableList<MigrationResult>> migratedEntries) {
+      this.untouchedEntries = untouchedEntries.collect(ImmutableList.toImmutableList());
+      this.validMigrations = migratedEntries.get("Valid");
+      this.invalidMigrations = migratedEntries.get("Invalid");
+    }
+
+    public ImmutableList<MigrationResult> getValidMigrations() {
```

Comment:
I believe this and the subsequent two public functions could use a one-liner of JavaDoc explaining what exactly they return.